src/components/image-upload/image-upload.tsx

```diff
@@ -0,0 +1,314 @@
+    }
+  }
+  const handlePickClick = () => {
+    if (imageInput.current !== null) {
+      imageInput.current.click();
+    }
+  }
+
+  const handleClose = () => {
+    setIsSubmitting(true);
+  };
+
+  const resetForm = () => {
+    if (formRef.current !== null) {
+      formRef.current.reset();
+    }
+    setPickedImage(undefined);
+    setAuthorCheckbox(true);
+    setFormData(initialFormState);
+    setFormError(INITIAL_ERROR_STATE);
+  };
+
+  const handleAuthorCheckboxClick = (value: boolean): void => {
+    setAuthorCheckbox(value);
+    if (user !== undefined) {
+      setFormData({ ...formData, author: user?.fullName });
+      setFormError((prevState) => {
+        const newState = {
+          ...prevState,
+          author: false
+        };
+        return newState;
+      });
+    }
+
+  }
+
+  const onFileUploadProgressChanged = (percentage: number) => {
+    setPercentage(percentage);
+  }
+
+  useEffect(() => setError(""), [logDialogIsOpen]);
+
+  useEffect(() => {
+    if (isSubmitting) {
+      setError("");
+      const createImage = async (): Promise<void> => {
+        try {
+          const fd = new FormData();
+          fd.append('author', formData.author);
+          fd.append('title', formData.title);
+          fd.append('image', formData.image as Blob);
+          fd.append('entityId', formData.entityId);
+          fd.append('entityType', formData.entityType);
+          await createImageAction(token ?? '', fd, onFileUploadProgressChanged);
+          setLogDialogIsOpen(false);
+          resetForm();
+
+          router.refresh();
+        } catch (error) {
+          setError("Prišlo je do napake pri shranjevanju fotografije.");
+          resetForm();
+        }
+        setIsSubmitting(false);
+      }
+
+      const file: File | null = formData.image;
+      const title: string = formData.title;
+      const formAuthor: string = formData.author;
+
+      if (file === null || title.length == 0 || (formAuthor !== null && formAuthor.length == 0)) {
+        setFormError((prevState) => {
+          const newState = {
+            ...prevState,
+            image: file === null,
+            title: title.length == 0,
+            author: formAuthor !== null && formAuthor.length == 0
+          };
+          return newState;
+        });
+        setIsSubmitting(false);
+      } else {
+        if (formRef.current !== null) {
+          formRef.current.reset();
+        }
+        if (imageInput.current) {
+          imageInput.current.value = "";
+        }
+
+        createImage();
+
+
+      }
+    }
+
+  }, [isSubmitting])
+
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    setIsSubmitting(true);
+    event.preventDefault();
+
+  }
+
+  const handleTitleChange = async (value: string) => {
+    if (titleRef.current !== null) {
+      setFormData({ ...formData, title: value });
+    }
+  }
+
+  const handleAuthorChange = async (value: string) => {
```

Comment:
does this need to be async? and why is titleRef relevant?

maybe the textfields will need to be adapted once main is merged again, because we moved them from aria to headless u.

Reply:
Fixed, thanks.